batchflow/dataset.py

```diff
@@ -33,24 +75,64 @@ def from_dataset(cls, dataset, index, batch_class=None):
 
     @staticmethod
     def build_index(index):
-        """ Create an index """
+        """ Create a DatasetIndex object from array-like data
+
+            Parameters
+            ----------
+            index : array-like
+
+            Returns
+            -------
+            index : DatasetIndex
+                DatasetIndex class object which was created from
+        """
         if isinstance(index, DatasetIndex):
             return index
         return DatasetIndex(index)
 
     @staticmethod
     def _is_same_index(index1, index2):
+        """ Check if index1 and index2 are equals
+
+            Parameters
+            ----------
+            index1 : array-like
```

Comment:
Not array-like, it is expected ot be an instance of DatasetIndex or its descendant